### ISSUE TITLE
Linking directly to an iron-icon element causes it to display as '}>'

### ISSUE THREAD

**Issue by smertens-npr** [vaadin/flow#8691]
### Versions:
    - Vaadin / Flow version:
    - Java version:
    - OS version:
    - Browser version (if applicable):
    - Application Server (if applicable):
    - IDE (if applicable):

**Comment by Legioth** [comment]
This happens because `@Id` causes runs the no-args constructor to create a Java instance. In the case of `Icon`, this sets a default icon which is the Vaadin logo. To fix this, we would first have to make it so that the constructor can realize that something else is already defined from the template: https://github.com/vaadin/flow/issues/3735

**Comment by smertens-npr** [comment]
Yes, it seems the problem is caused by pushing state when you don't know what you're pushing state into.  If it were possible for an iron-icon to exist without the icon attribute being defined, then there would be no need to push state from the constructor.

This is admittedly a very small problem with an easy workaround.  If architecture design makes a real fix impractical, then I suggest at least documenting it in the java doc so developers aren't confused.

**Comment by denis-anisimov** [comment]
The Flow part is fixed now: the template value is set on the server side into the `Element`.
Now the constructor implementation of `Icon` should check whether the `Element` already has the attribute/property set before setting the default value (as it's done for may components already by me for the "value" property).
